### PR TITLE
CVE-2010-4494 and CVE-2011-2835

### PR DESCRIPTION
**Pull request by jackfrate**

### CODE REVIEW

File: cves/CVE-2010-4494.yml [hunk 1]
```diff
@@ -4,14 +4,14 @@ CWE_instructions: |
   Please go to cwe.mitre.org and find the most specific, appropriate CWE entry
   that describes your vulnerability. (Tip: this may not be a good one to start
   with - spend time understanding this vulnerability before making your choice!)
-CWE: 
+CWE: "https://cwe.mitre.org/data/definitions/643.html"
```
Comment (andymeneely) [inline cves/CVE-2010-4494.yml]: Just the number is fine - we pull in the info ourselves.

File: cves/CVE-2010-4494.yml [hunk 2]
```diff
@@ -4,14 +4,14 @@ CWE_instructions: |
...
 curated_instructions: |
   If you are manually editing this file, then you are "curating" it. Set the
```
Comment (andymeneely) [inline cves/CVE-2010-4494.yml]: Make sure you name the PR like the others are named

File: cves/CVE-2011-2835.yml [hunk 3]
```diff
@@ -107,20 +110,20 @@ discovered:
...
   question: |
     What subsystems was the mistake in?
 
     Look at the path of the source code files code that were fixed to get
     directory names. Look at comments in the code. Look at the bug reports how
     the bug report was tagged. Examples: "clipboard", "gpu", "ssl", "speech", "renderer"
-  answer: 
-  name: 
+  answer: Network internals
+  name:
```
Comment (CatherineOsadciw) [inline cves/CVE-2011-2835.yml]: Maybe put "Cache" for subsystem name?

File: cves/CVE-2011-2835.yml [hunk 4]
```diff
@@ -132,10 +135,10 @@ interesting_commits:
     If there are no interesting commits, demonstrate that you completed this section by explaining what happened between the VCCs and the fix.
   answer:
   commits:
-  - commit: 
-    note: 
-  - commit: 
-    note: 
+    - commit:
```
Comment (CatherineOsadciw) [inline cves/CVE-2011-2835.yml]: Demonstrate that you completed this section by explaining what happened between the VCCs and the fix.

File: cves/CVE-2011-2835.yml [hunk 5]
```diff
@@ -210,4 +213,6 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer: 
+  answer: |
+    I believe that the main mistake was improper knowledge of the systems. By using
```
Comment (CatherineOsadciw) [inline cves/CVE-2011-2835.yml]: Add some more details to your analysis

File: cves/CVE-2010-4494.yml [hunk 6]
```diff
@@ -32,43 +32,45 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: "There is a double free vulnerability in libxml2 2.7.8 and other version
-  as used in gogole chrome before 8.0.552.215.\n\nDouble free vulnerability is a common
+description:
+  "There is a double free vulnerability in libxml2 2.7.8 and other version
+  as used in gogole chrome before 8.0.552.215.\nThe vulnerability allows one to pop an XPath object off the stack, and
```
Comment (kzalbak) [inline cves/CVE-2010-4494.yml]: You have some "\n" characters in this description, you might be able to take these out to make it easier to read.

File: cves/CVE-2010-4494.yml [hunk 7]
```diff
@@ -212,10 +218,9 @@ mistakes:
...
-    mistakes is the main reason that lead to this vulnerability. \nThe developer shall
-    put some if statement to check an object before he free the object. The file contains
-    more than ten thousands line of code, so it is neccessary to keep track of the
-    process of using a new object and free an old object\nI believe that the vulnerability
+  answer:
+    "Double free vulnerability is a common memory corruption error if the developers
+    don't write a proper check case before they free memory\nIn this case, it seems like
+    a developer simply forgot or was not aware of this. It can also be very hard to catch,
+    as the chromium code base in incredibly large, and can not be easily combed through.\nI believe that the vulnerability
     was fixed correctly.\n"
```
Comment (kzalbak) [inline cves/CVE-2010-4494.yml]: Why do you think the vulnerability was fixed correctly? What was done in the proper manner, and what exactly was done to prevent this issue?

File: cves/CVE-2011-2835.yml [hunk 8]
```diff
@@ -210,4 +213,6 @@ mistakes:
...
+    a raw certificate, it caused race conditions that in the end were easily mitigated.
```
Comment (kzalbak) [inline cves/CVE-2011-2835.yml]: How were the race conditions mitigated? What fix lead to this?

File: cves/CVE-2010-4494.yml [hunk 9]
```diff
@@ -143,12 +147,12 @@ major_events:
...
+  answer: no major events observed.
   events:
-  - name: 
-    date: 
-  - name: 
-    date: 
+    - name:
+      date:
+    - name:
+      date:
```
Comment (ethandellaposta) [inline cves/CVE-2010-4494.yml]: Remove unnecessary lines

File: cves/CVE-2010-4494.yml [hunk 10]
```diff
@@ -212,10 +218,9 @@ mistakes:
...
-    don't write a proper check case before they free memory\nIn this case, the coding
-    mistakes is the main reason that lead to this vulnerability. \nThe developer shall
-    put some if statement to check an object before he free the object. The file contains
-    more than ten thousands line of code, so it is neccessary to keep track of the
-    process of using a new object and free an old object\nI believe that the vulnerability
+  answer:
+    "Double free vulnerability is a common memory corruption error if the developers
+    don't write a proper check case before they free memory\nIn this case, it seems like
+    a developer simply forgot or was not aware of this. It can also be very hard to catch,
+    as the chromium code base in incredibly large, and can not be easily combed through.\nI believe that the vulnerability
```
Comment (ethandellaposta) [inline cves/CVE-2010-4494.yml]: Why the quotes?

File: cves/CVE-2010-4494.yml [hunk 11]
```diff
@@ -101,9 +103,10 @@ discovered:
 
     If there is no evidence as to how this vulnerability was found, then you may
     leave the entries blank except for "answer". Write down where you looked in "answer".
-  answer: "The vulnerability was discovered by a user and he reported it to the dev
-    team. \nThere was an instruction to reproduce the vulnerability in multiple version
-    of chrome\n"
+  answer:
+    "The vulnerability was discovered by a user and he reported it to the dev
+    team. \nSince they could not easily clone the entire repo, they left instructions for developers who could for
+    multiple versions of chrome.\n"
```
Comment (ethandellaposta) [inline cves/CVE-2010-4494.yml]: \n and weird return formatting.

File: cves/CVE-2010-4494.yml (hunk 1 above)
Comment (ethandellaposta) [inline cves/CVE-2010-4494.yml]: 2 upvotes

File: cves/CVE-2011-2835.yml [hunk 12]
```diff
@@ -4,14 +4,14 @@ CWE_instructions: |
   Please go to cwe.mitre.org and find the most specific, appropriate CWE entry
   that describes your vulnerability. (Tip: this may not be a good one to start
   with - spend time understanding this vulnerability before making your choice!)
-CWE: 
+CWE: https://cwe.mitre.org/data/definitions/362.html
```
Comment (ethandellaposta) [inline cves/CVE-2011-2835.yml]: 2 upvotes

File: cves/CVE-2010-4494.yml [hunk 13]
```diff
@@ -32,43 +32,45 @@ description_instructions: |
...
+  - 63444
+repo:
 fixes_vcc_instructions: |
   Please put the commit hash in "commit" below (see my example in
   CVE-2011-3092.yml). Fixes and VCCs follow the same format.
 fixes:
-- :commit: 4b41418a14d224fcc3dcd8b1b5923920ea709d5c
-  :note: fix Xpath bug from upstream
+  - :commit: 4b41418a14d224fcc3dcd8b1b5923920ea709d5c
+    :note: fix Xpath bug from upstream
```
Comment (Mirldog) [inline cves/CVE-2010-4494.yml]: It might be helpful to include specifics about what happens in this commit, beyond just the outcome

File: cves/CVE-2010-4494.yml [hunk 14]
```diff
@@ -115,8 +118,9 @@ subsystem:
     Look at the path of the source code files code that were fixed to get
     directory names. Look at comments in the code. Look at the bug reports how
     the bug report was tagged. Examples: "clipboard", "gpu", "ssl", "speech", "renderer"
-  answer: This is XML Path language implementation in to address part of an XML document.
-    It was desinged to be used by both XSLT and XPointer
+  answer:
+    This is XML Path language implementation in to address part of an XML document.
+    It was designed to be used by both XSLT and XPointer, but because of this an object could be freed twice.
```
Comment (Mirldog) [inline cves/CVE-2010-4494.yml]: Does that mean this was a race condition between two subsystems using the same resource? Could be interesting to elaborate on

File: cves/CVE-2010-4494.yml [hunk 15]
```diff
@@ -129,12 +133,12 @@ interesting_commits:
...
-  - commit: 681c85692a6421e1f288c329f80dcee25355dd7b
-    note: |
-      Refactoring the project was performed by moved all upstream code into a src/ subdirectory.
-      This helps keep all patches, build files, and metadata separate.
-  - commit: 6999dc89b49b7424f222e6888955d08efce52dde
-    note: remove non working functions on os object and d8 window. i
+    - commit: 681c85692a6421e1f288c329f80dcee25355dd7b
+      note: |
+        Refactoring the project was performed by moved all upstream code into a src/ subdirectory.
+        This helps keep all patches, build files, and metadata separate.
```
Comment (Mirldog) [inline cves/CVE-2010-4494.yml]: Just a grammar thing, you probably meant 'by moving' instead of 'by moved'

File: cves/CVE-2010-4494.yml [hunk 16]
```diff
@@ -166,40 +170,42 @@ lessons:
     free to give it a small name and add one in the same format as these.
   defense_in_depth:
     applies: true
-    note: "The fix for this vulnerability involved putting an if statement to check
+    note:
+      "The fix for this vulnerability involved putting an if statement to check
       on context object. \nThe developers plan to avoid to catch one of the few case
       to free the same memory twice. \n"
```
Comment (Mirldog) [inline cves/CVE-2010-4494.yml]: fix \n issues

File: cves/CVE-2011-2835.yml [hunk 17]
```diff
@@ -32,51 +32,54 @@ description_instructions: |
...
+  - :commit: 72f508123a618d0ca9051474990658b8180e0049
+    :note: "Now it will cache the os specific certificate rather than the raw certificate."
+  - :commit: 66a2bd787770beee5a55f6dfd9ba5cdaa1cc91b1
+    :note: "just re-enabling tests now that it is fixed"
+  - :commit: 40a089c802ae7e4e514d2228c84c115842c4d371
+    :note: "After the fix there was an edge case with a possible vulnerability"
+  - :commit: ebaf26ce325e21df4619ab95cc0399bc88f30d05
+    :note: "Since there is a problem again now, they had to disable a test"
+  - :commit: ee16560ee3e913b6aaa8bb8eb6f7beb14c85bb5a
+    :note: ""
```
[the remark this answers is not in the document]
Comment (jackfrate) [reply cves/CVE-2011-2835.yml]: probably miscommunication
